ultratrace2/model/tests/test_xhair.py

```diff
@@ -0,0 +1,52 @@
+import math
+
+import pytest
+
+from ..xhair import XHair
+from ..color import Color
+
+
+def _get_black() -> Color:
+    return Color(0, 0, 0)
+
+
+@pytest.mark.parametrize(
+    "p,q,expected",
+    [
+        (XHair(_get_black, 0, 0), XHair(_get_black, 1, 0), 1),
+        (XHair(_get_black, 0, 0), XHair(_get_black, 1, 1), 2),
```

Comment:
Maybe we should add a test that actually squares stuff :^)
```suggestion
        (XHair(_get_black, 0, 0), XHair(_get_black, 1, 1), 2),
        (XHair(_get_black, 2, 0), XHair(_get_black, 0, 2), 8),
```

Reply:
a07a353